test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -240,49 +244,69 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
 
       stub_policy current_user, api_user, update?: false
 
-      patch :update, params: { api_user_id: api_user, application_id: application, application: { supported_permission_ids: [] } }
+      patch :update, params: {
+        api_user_id: api_user,
+        application_id: application,
+        application: { supported_permission_ids: [] },
+      }
 
       assert_not_authorised
     end
 
     should "push permission changes out to apps" do
-      sign_in create(:superadmin_user)
+      application = create(:application)
+      permission = create(:supported_permission, application:)
 
-      application = create(:application, with_non_delegatable_supported_permissions: %w[permission])
-      api_user = create(:api_user)
-      create(:access_token, resource_owner_id: api_user.id, application:)
+      api_user = create(:api_user, with_signin_permissions_for: [application])
+      create(:access_token, application:, resource_owner_id: api_user.id)
 
-      permission = application.supported_permissions.find_by(name: "permission")
+      sign_in create(:superadmin_user)
 
       PermissionUpdater.expects(:perform_on).with(api_user)
 
-      patch :update, params: { api_user_id: api_user, application_id: application, application: { supported_permission_ids: [permission.id] } }
+      patch :update, params: {
+        api_user_id: api_user,
+        application_id: application,
+        application: { supported_permission_ids: [permission.id] },
+      }
     end
 
-    should "exclude applications with revoked access tokens" do
-      sign_in create(:superadmin_user)
-
+    should "prevent updating permissions if the user has a revoked access token" do
```

Comment:
Could this description be more explicit in how it contrasts with the next one? E.g. "user _only_ has a revoked access token". Or maybe there could be a shared context:

```rb
context "when access tokens for the application are revoked" do
  context "and there are no non-revoked access tokens for the application" do
  context "but there are also non-revoked access tokens for the application" do
```